src/main/java/org/hydev/mcpm/client/installer/presenter/InstallPresenter.java

```diff
@@ -0,0 +1,27 @@
+package org.hydev.mcpm.client.installer.presenter;
+
+import org.hydev.mcpm.client.installer.InstallResult;
+
+import java.util.function.Consumer;
+
+/**
+ * Implementation to the ResultPresenter, display the result of installation of plugins
+ */
+public class InstallPresenter implements ResultPresenter {
+
+    private Consumer<String> log;
+    /**
+     * Instantiate Install Presenter
+     *
+     * @param log log string to the console
+     */
+
+    public InstallPresenter(Consumer<String> log) {
+        this.log = log;
+    }
+
+    @Override
+    public void displayResult(String message) {
+        this.log.accept(message);
```

Comment:
This is pretty bare bones. I would recommend returning an InstallResult object with everything you need to print. Let the presenter turn that into a nice string!

Reply:
Gotcha!